gems/smithy/lib/smithy/generators/client.rb

```diff
@@ -11,41 +11,67 @@ def initialize(plan)
         super
       end
 
-      # @return [Enumerator<String, String>] The file paths and their contents to generate.
       def generate
-        files = source_files
-        files.each do |file, content|
+        gem_files.each_with_object([]) do |(file, content), files|
           next if file == "lib/#{@gem_name}/customizations.rb" && should_skip_customizations?
 
           create_file file, content
+          files << file
         end
-        files
+      end
+
+      def source
+        source = source_files.map { |_file, content| content }.join("\n")
+        Object.module_eval(source)
```

Comment:
I would argue that generators should never be responsible for evaluating ruby code or getting the module.  Just generating the source.

I think something like the client helper should be responsible for evaluating code/returning the module.

Reply:
Yeah, I went back and forth on this. On one hand it's nice to just be returned a module that is the loaded code, on the other, given a string, the user will have to load themselves (or us in the helper file). Let me think through the implications.

Reply:
One downside of string only is we don't know the resolved namespace.